### ISSUE TITLE
Encoding 1252 data could not be found. Make sure you have correct international codeset assembly installed and enabled.

### ISSUE THREAD

**Issue by Scall** [ClassicUO/ClassicUO#715]
**Issue**
The client crashes with the following error:

`[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'ClassicUO.Utility.StringHelper' threw an exception. ---> System.NotSupportedException: Encoding 1252 data could not be found. Make sure you have correct international codeset assembly installed and enabled.
  at System.Text.Encoding.GetEncoding (System.Int32 codepage) [0x0023f] in <7768781867c04892b60eb12ebb814c26>:0 
  at ClassicUO.Utility.StringHelper..cctor () [0x00020] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
   --- End of inner exception stack trace ---
  at ClassicUO.Network.Packet.ReadASCII (System.Int32 length, System.Boolean exitIfNull) [0x00049] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at ClassicUO.Game.Scenes.ServerListEntry..ctor (ClassicUO.Network.Packet reader) [0x00012] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at ClassicUO.Game.Scenes.LoginScene.ParseServerList (ClassicUO.Network.Packet reader) [0x00024] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at ClassicUO.Game.Scenes.LoginScene.NetClient_PacketReceived (System.Object sender, ClassicUO.Network.Packet e) [0x00068] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at (wrapper delegate-invoke) System.EventHandler`1[ClassicUO.Network.Packet].invoke_void_object_TEventArgs(object,ClassicUO.Network.Packet)
  at ClassicUO.Utility.Exstentions.Raise[T] (System.EventHandler`1[TEventArgs] handler, T e, System.Object sender) [0x00003] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at ClassicUO.Network.NetClient.Update () [0x0002f] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at ClassicUO.Engine.OnNetworkUpdate (System.Double totalMS, System.Double frameMS) [0x00034] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at ClassicUO.Engine.Tick () [0x00034] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at Microsoft.Xna.Framework.SDL2_FNAPlatform.RunLoop (Microsoft.Xna.Framework.Game game) [0x006a9] in <e7e84f10a147490c97cb0c9637334a11>:0 
  at Microsoft.Xna.Framework.Game.Run () [0x0002c] in <e7e84f10a147490c97cb0c9637334a11>:0 
  at ClassicUO.Engine.Run (System.String[] args) [0x00019] in <a031f9cf4bd140dcb8a01aa279e33420>:0 
  at ClassicUO.Bootstrap.Main (System.String[] args) [0x00028] in <a031f9cf4bd140dcb8a01aa279e33420>:0 `

This is because it only tries to load I18N.dll in "/usr/**local**/lib/mono" (/usr/local/lib/mono/gac/I18N/4.0.0.0__0738eb9f132ed756/I18N.dll), where it may not be found, instead of trying in "/usr/lib/mono" as well.

Basically:
/usr/**local**/lib/mono -> Canonical path for those who built Mono by themselves -> not affected by this issue.
/usr/lib/mono -> Canonical path for those who uses the Mono package provided by their Linux distribution -> **affected**.

**Steps to reproduce the issue**
1. Run the ClassicUO **dev-preview** build on Linux, with Mono **not** installed in "/usr/local".
2. Try to log in.

**Note**: this issue is not present in ClassicUO Beta 0.1.2.0.

**Workaround**
Create a symbolic link in "/usr/local/lib" pointing to "/usr/lib/mono", like this:

`ln -s /usr/lib/mono /usr/local/lib/mono`

**Comment by andreakarasho** [comment]
@SaschaKP

**Comment by SaschaKP** [comment]
this seems more a mono bug, than a cuo bug. (or possibly a distribution related bug, not anyway something we can handle globally)

This can't be handled as a bug through CUO, because it's a locality happening in a distribution, and is something present only locally, I've tested with gentoo, and of course it works (through it's not a correct test, since gentoo is locally compiled on install)

what distribution are you using?
have you tried with unsafe mono install package (usually major distribution don't install latest mono, but safest one, as I remember they are beyond by a quite number from the mainstream mono, on some distros)

Possible solution would be to include the codepages dll in the cuo package, but I don't know if this would affect someone else or if it will work in those particular cases.

We should mark this issue as won't fix anyway, and in case some users happen to have same problem, in case of same distro, mark to the distro owner the problem, in case of different distro, mark to mono the problem (even if the latest one seems the less probable, although possible).

**Comment by Scall** [comment]
I can reproduce the issue with both Gentoo (Mono 6.0.0.334) and Arch Linux (Mono 6.0.0.327), but I've just discovered that it only occurs when launching the client through the shell script "`ClassicUO`" (I always used it without issues with ClassicUO Beta 0.1.2.0). The issue does **not** occur when launching the client with the command "`mono ClassicUO.exe`".

**Comment by andreakarasho** [comment]
I think it happens due to the old selfrunning mono version distribuited with cuo
